src/main/java/gui/PaintPanel.java

```diff
@@ -80,17 +81,21 @@ protected void paintComponent(Graphics g) {
         resizedIcon.paintIcon(this,graphics2D,0,0);
 
         //Edges
+        ArrayList<EdgeWithCoordinates> edges = new ArrayList<>();
         try{
-            for (EdgeWithCoordinates e : edgeWithCoordinates){
-                int [] scaleFrom = scaleCoords(e.getFromX(),e.getFromY(),5);
-                int [] scaleTo = scaleCoords(e.getToX(),e.getToY(),5);
+            for (int i = 0; i < edgeWithCoordinates.size(); i++){
```

Comment:
Zamień to na styl edgesWithCoordinates albo coordinatedEdges, bo aktualna nazwa listy wskazuje na pojedynczy obiekt.